jupyterhub/auth.py

```diff
@@ -902,23 +1001,25 @@ def _add_user_cmd_default(self):
         help="""
         Allow login from all users in these UNIX groups.
 
-        If set, allowed username set is ignored.
+        .. versionchanged:: 5.0
+            `allowed_groups` may be specified together with allowed_users,
+            to grant access by group OR name.
         """
     ).tag(config=True)
 
-    @observe('allowed_groups')
-    def _allowed_groups_changed(self, change):
-        """Log a warning if mutually exclusive user and group allowed sets are specified."""
-        if self.allowed_users:
-            self.log.warning(
-                "Ignoring Authenticator.allowed_users set because Authenticator.allowed_groups supplied!"
-            )
+    @default("allow_all")
+    def _allow_all_default(self):
+        if self.allowed_users or self.allowed_groups:
+            # if any allow config is specified, default to False
```

Comment:
Should include `allow_existing_users` also, right? Discussed also in https://github.com/jupyterhub/jupyterhub/pull/4701/files#r1531617284